Python/dynamic/household.py

```diff
@@ -19,38 +21,71 @@
+        ability_weights = ability percentile groups (Jx1 array)
+        g_n = population growth rate (scalar)
+    Output:
+        K_now = Aggregate Capital (scalar)
     '''
-    K_now = np.sum(b * weights)
+    K_now = np.sum(b * pop_weights * ability_weights)
+    K_now /= 1.0 + g_n
     return K_now
 
 
-def marg_ut_cons(c, params):
+def get_BQ(r, b_splus1, pop_weights, ability_weights, rho, g_n):
     '''
-    Inputs: Consumption, parameters
+    Inputs:
+        r = interest rate (scalar)
+        b_splus1 = wealth holdings for the next period (SxJ array)
+        pop_weights = population weights by age (Sx1 array)
+        ability_weights = ability weights (Jx1 array)
+        rho = mortality rates (Sx1 array)
+        g_n = population growth rate (scalar)
+    Output:
+        BQ = aggregate BQ (Jx1 array)
+    '''
+    BQ = (1+r) * (b_splus1 * pop_weights * rho).sum(0) * ability_weights
+    BQ /= 1.0 + g_n
+    return BQ
+
 
-    Returns:    Marginal Utility of Consumption
+def marg_ut_cons(c, params):
+    '''
+    Inputs:
+        c = Consumption (any array or scalar)
+        params = list of parameters (list)
+    Outputs:
+        output = Marginal Utility of Consumption (same shape as c)
     '''
-    J, S, T, beta, sigma, alpha, Z, delta, ltilde, nu, g_y, tau_payroll, retire, mean_income_data, a_tax_income, b_tax_income, c_tax_income, d_tax_income, h_wealth, p_wealth, m_wealth, b_ellipse, upsilon = params
+    J, S, T, beta, sigma, alpha, Z, delta, ltilde, nu, g_y, g_n_ss, tau_payroll, retire, mean_income_data, a_tax_income, b_tax_income, c_tax_income, d_tax_income, h_wealth, p_wealth, m_wealth, b_ellipse, upsilon = params
     output = c**(-sigma)
     return output
 
 
 def marg_ut_labor(n, chi_n, params):
+   '''
+    Inputs:
+        n = labor particpation distribution (various length array or scalar)
+        chi_n = chi^n_s (various length array or scalar)
+        params = list of parameters (list)
+    Output:
+        output = Marginal Utility of Labor (same shape as n)
     '''
-    Inputs: Labor, chi^n_s, parameters
-
-    Returns:    Marginal Utility of Labor
-    '''
-    J, S, T, beta, sigma, alpha, Z, delta, ltilde, nu, g_y, tau_payroll, retire, mean_income_data, a_tax_income, b_tax_income, c_tax_income, d_tax_income, h_wealth, p_wealth, m_wealth, b_ellipse, upsilon = params
+    J, S, T, beta, sigma, alpha, Z, delta, ltilde, nu, g_y, g_n_ss, tau_payroll, retire, mean_income_data, a_tax_income, b_tax_income, c_tax_income, d_tax_income, h_wealth, p_wealth, m_wealth, b_ellipse, upsilon = params
     try:
-        deriv = b_ellipse * (1/ltilde) * ((1 - (n / ltilde) ** upsilon) ** (
-            (1/upsilon)-1)) * (n / ltilde) ** (upsilon - 1)
+        deriv = b_ellipse * (1.0/ltilde) * ((1.0 - (n / ltilde) ** upsilon) ** (
+            (1.0/upsilon)-1.0)) * (n / ltilde) ** (upsilon - 1.0)
     except ValueError as ve:
+        # I think TJ added this in.  We need to be careful with doing stuff like this --
```

Comment:
Yes, this was done in haste to get the small test to pass. I considered a few options here. One thing that seemed reasonable was to have a `permissive` flag (or something similar) that would catch certain exceptions and provide some small (or large) value instead of just raising the exception. In 'real' code, this flag would be False, so that the exception would not be caught. In the "smoke test", we would allow the exception to be caught. Just an idea.